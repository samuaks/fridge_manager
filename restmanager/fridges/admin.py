from django.contrib import admin
from .models import Fridge
# Register your models here.
class AppAdmin(admin.ModelAdmin):
    list_display = ('name', 'state', 'floor')

admin.site.register(Fridge, AppAdmin)