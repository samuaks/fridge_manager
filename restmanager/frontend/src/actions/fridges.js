import axios from "axios";

import { GET_FRIDGES, DELETE_FRIDGE, ADD_FRIDGE, BOT_NOTIFY } from "./types";

// get fridges
export const getFridges = () => dispatch => {
  axios
    .get("/api/fridges/")
    .then(res => {
      dispatch({
        type: GET_FRIDGES,
        payload: res.data
      });
    })
    .catch(err => console.log(err));
};

// delete fridges
export const deleteFridge = id => dispatch => {
  axios
    .delete(`/api/fridges/${id}/`)
    .then(res => {
      dispatch({
        type: DELETE_FRIDGE,
        payload: id
      });
    })
    .catch(err => console.log(err));
};

// add fridge
export const addFridge = fridge => dispatch => {
  axios
    .post("/api/fridges/", fridge)
    .then(res => {
      dispatch({
        type: ADD_FRIDGE,
        payload: res.data
      });
    })
    .catch(err => console.log(err));
};
//general notify function to #general in test_slack_env
export function botNotification() {
  axios.get("/api/beer").then(function(response) {
    console.log(response);
  });
}

export const botNotify = id => dispatch => {
  axios
    .get(`/api/fridges/${id}/`)
    .then(res => {
      dispatch({
        type: BOT_NOTIFY,
        payload: id
      });
    })
    .catch(err => console.log(err));
};
