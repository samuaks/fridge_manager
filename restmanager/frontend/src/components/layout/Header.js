import React, { Component } from "react";

export class Header extends Component {
  render() {
    /*
        Header from https://getbootstrap.com/docs/4.1/components/navbar/
        */
    return (
      <nav className="navbar navbar-expand-sm navbar-light bg-light">
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarTogglerDemo01"
          aria-controls="navbarTogglerDemo01"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarTogglerDemo01">
          <a className="navbar-brand" href="#">
            Fridge Management UI
          </a>
          <ul className="navbar-nav mr-auto mt-2 mt-lg-0"></ul>
        </div>
      </nav>
    );
  }
}

export default Header;
