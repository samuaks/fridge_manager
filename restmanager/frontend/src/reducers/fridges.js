import { GET_FRIDGES, DELETE_FRIDGE, ADD_FRIDGE } from "../actions/types.js";

const initialState = {
  fridges: []
};

export default function(state = initialState, action) {
  switch (action.type) {
    case GET_FRIDGES:
      return {
        ...state,
        fridges: action.payload
      };
    case DELETE_FRIDGE:
      return {
        ...state,
        fridges: state.fridges.filter(fridge => fridge.id !== action.payload)
      };
    case ADD_FRIDGE:
      return {
        ...state,
        fridge: [...state.fridges, action.payload]
      };

    default:
      return state;
  }
}
