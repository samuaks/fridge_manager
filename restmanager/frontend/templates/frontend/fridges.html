<!DOCTYPE html>
<html lang="en">
    <head>
        <style>
            .box{
                border:2px solid #000000;
            }

            a:link{
                font-family:Segoe UI;
                font-size:18px;
                font-weight:bold;
                color:#000000;
            }

            a:active{
                font-family:Segoe UI;
                font-size:18px;
                font-weight:bold;
                color:#000000;
            }

            .button_1 {
                background-color:#ffffff;
                border-radius:5px;
                border:2px solid #5c5b5c;
                display:inline-block;
                cursor:pointer;
                color:#000000;
                font-family:BlinkMacSystemFont;
                font-size:18px;
                padding:6px 24px;
            }

            .button_1:hover {
                background-color:#4aa48a;
            }

            .button_1:active {
                position:relative;
                top:1px;
            }

            .container {
                text-align: center;
            }

            .center {
                text-align: center;
                width: auto;
                height: auto;
                display: inline-block;
                position: relative;
            }
        </style>

        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        {% load static %}
        <link rel="stylesheet" href="https://www.w3schools.com/w3css/4/w3.css"/>
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bulma/0.6.2/css/bulma.css"/>
        <link rel="stylesheet" href="https://bootswatch.com/4/minty/bootstrap.min.css"/>
        <title>Fridges</title>
    </head>

    <body>
        <div class="container">
            <div class="center">
                <a href="/fridges/"><p>Back to all fridges</p></a>
                <a href="/"><p>Back to floor select</p></a>
                {% for ctx in data %}
                <div id="items-wrapper" class="box">
                    <div class="media-left">
                        <div class="container">

                            <p hidden>ID: {{ctx.id}}</p>
                            <a href="/fridges/?id={{ctx.id}}"><h1>{{ctx.name}}</h1></a>
                            <a href="/fridges/?state={{ctx.state}}"><h4>State: {{ctx.state}}</h4></a>
                            <a href="/fridges/?floor={{ctx.floor}}"><h4>Floor: {{ctx.floor}}</h4></a>

                            <form action="/api/change_state/" id="post_form" method="post">
                                <input type="hidden" name="id" id="id" value="{{ctx.id}}">
                                <input type="hidden" name="name" id="name" value="{{ctx.name}}">
                                <input type="hidden" name="state" id="state" value="{{ctx.state}}">
                                <input type="hidden" name="floor" id="floor" value="{{ctx.floor}}">
                                {%  with curr_state=ctx.state %}
                                    {%  if curr_state == "Empty" %}
                                        <input type="submit" class="button_1" value="Change state to Full">
                                    {% elif curr_state == "Full" %}
                                        <input type="submit" class="button_1" value="Change state to Half-full">
                                    {% else %}
                                        <input type="submit" class="button_1" value="Change state to Empty">
                                    {% endif %}
                                {% endwith %}
                            </form>
                        </div>
                    </div>
                </div>
                {% endfor %}
            </div>
        </div>
    </body>
</html>
