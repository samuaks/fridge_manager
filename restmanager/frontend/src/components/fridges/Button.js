import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { getFridges, deleteFridge } from "../../actions/fridges";

export class Fridges extends Component {
  static propTypes = {
    fridges: PropTypes.array.isRequired,
    getFridges: PropTypes.func.isRequired,
    deleteFridge: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.props.getFridges();
  }

  render() {
    return (
      <Fragment>
        <h2>Fridges</h2>
        <table className="table table-striped">
          <thead>
            <tr>
              <th>ID</th>
              <th>Name</th>
              <th>STATUS</th>
              <th>Time Since</th>
              <th></th>
              <th></th>
            </tr>
          </thead>
          <tbody>
            {this.props.fridges.map(fridge => (
              <tr key={fridge.id}>
                <td>{fridge.id}</td>
                <td>{fridge.name}</td>
                <td>{fridge.fridge_is_empty ? "EMPTY" : "NOT EMPTY"}</td>
                <td>{fridge.time_since.split("T")[0]}</td>
                <td>
                  <button
                    onClick={this.props.deleteFridge.bind(this, fridge.id)}
                    className="btn btn-danger btn-sm"
                  >
                    Delete
                  </button>
                </td>
                <td>
                  <button className="btn btn-success btn-sm">Notify</button>
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      </Fragment>
    );
  }
}

const mapStatetoProps = state => ({
  fridges: state.fridges.fridges
});

export default connect(mapStatetoProps, { getFridges, deleteFridge })(Fridges);
