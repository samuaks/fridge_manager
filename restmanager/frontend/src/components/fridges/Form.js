import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { addFridge } from "../../actions/fridges";

export class Form extends Component {
  state = {
    name: "",
    fridge_is_empty: false
  };

  static propTypes = {
    addFridge: PropTypes.func.isRequired
  };

  handleFridgeStatus = e => {
    const fridge_is_empty = e.target.checked;
    this.setState({ fridge_is_empty });
  };

  onChange = e => this.setState({ [e.target.name]: e.target.value });

  onSubmit = e => {
    e.preventDefault();
    const { name, fridge_is_empty } = this.state;
    const fridge = { name, fridge_is_empty };
    this.props.addFridge(fridge);
  };

  render() {
    const { name, fridge_is_empty } = this.state;
    return (
      <div className="card card-body mt-4 mb-4">
        <h2>Add Lead</h2>
        <form onSubmit={this.onSubmit}>
          <div className="form-group">
            <label>Name</label>
            <input
              className="form-control"
              type="text"
              name="name"
              onChange={this.onChange}
              value={name}
            />
          </div>
          <div className="form-group">
            <label>Fridge is empty</label>
            <input
              name="fridge_is_empty"
              type="checkbox"
              checked={fridge_is_empty}
              onChange={this.handleFridgeStatus}
            />
          </div>
          <div className="form-group">
            <button type="submit" className="btn btn-primary">
              Submit
            </button>
          </div>
        </form>
      </div>
    );
  }
}
//Since we dont need to bring any state back to this component,
// so we can have "null" as first parameter and only include the method that we are calling
export default connect(null, { addFridge })(Form);
