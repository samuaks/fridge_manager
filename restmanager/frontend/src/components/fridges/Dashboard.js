import React, { Fragment } from "react";
import Form from "./Form";
import Fridges from "./Fridges";

/*
Dasboard includes:
Form.js
Fridges.js 
*/

export default function Dashboard() {
  return (
    <Fragment>
      <Form />
      <Fridges />
    </Fragment>
  );
}
