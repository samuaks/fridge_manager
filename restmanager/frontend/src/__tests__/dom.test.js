import React from 'react';
import Header from '../components/layout/Header';
import {create, act} from 'react-test-renderer';
import fridges from '../assests/fridges-mock';

/*
let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
});

afterEach(() => {
  // cleanup on exiting
  container.remove();
  container = null;
});
*/

it('dashboard renders correctly', () => {
    let layout = null;
    act(() => {
        layout = create(<Header />);
    });
    expect(layout.toJSON()).toMatchSnapshot();
    layout.unmount();
});