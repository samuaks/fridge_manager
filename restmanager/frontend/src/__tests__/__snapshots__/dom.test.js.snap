// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard renders correctly 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-light bg-light"
>
  <button
    aria-controls="navbarTogglerDemo01"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarTogglerDemo01"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarTogglerDemo01"
  >
    <a
      className="navbar-brand"
      href="#"
      id="link-render-test"
    >
      Fridge Management UI
    </a>
    <ul
      className="navbar-nav mr-auto mt-2 mt-lg-0"
    />
  </div>
</nav>
`;
